packages/client/services/sort_service.ts

```diff
@@ -0,0 +1,84 @@
+import { Transaction } from "@cryptify/common/src/domain/entities/transaction";
+
+function sort_date_newest(transactions: Transaction[]) {
+    return transactions.sort((a, b) => new Date(b.createdAt).valueOf() - new Date(a.createdAt).valueOf());
+}
+
+function sort_date_oldest(transactions: Transaction[]) {
+    return transactions.sort((a, b) => new Date(a.createdAt).valueOf() - new Date(b.createdAt).valueOf());
+}
+
+function sort_amount_highest(transactions: Transaction[], walletAddress: string) {
+    // Filters the array of transactions based on if theyre deposits or withdrawls
+    // Performs the sort on the two shallow copy arrays and then concats the two arrays
+
+    return transactions
+        .filter((transaction) => transaction.walletIn === walletAddress)
+        .sort((a, b) => Number(b.amount) - Number(a.amount))
+        .concat(
+            transactions
+                .filter((transaction) => transaction.walletIn !== walletAddress)
+                .sort((a, b) => Number(a.amount) - Number(b.amount)),
+        );
+}
+
+function sort_amount_lowest(transactions: Transaction[], walletAddress: string) {
+    // Functions the same as described above
+
+    return transactions
+        .filter((transaction) => transaction.walletIn !== walletAddress)
+        .sort((a, b) => Number(b.amount) - Number(a.amount))
+        .concat(
+            transactions
+                .filter((transaction) => transaction.walletIn === walletAddress)
+                .sort((a, b) => Number(a.amount) - Number(b.amount)),
+        );
+}
+
+// Applying the sort functions based on which sort option is selected
+function sort_Transactions(sortType: string, transactions: Transaction[], walletAddress: string): Transaction[] {
+    if (sortType === "sortDateNewest") {
+        return sort_date_newest([...transactions]);
+    }
+    if (sortType === "sortDateOldest") {
+        return sort_date_oldest([...transactions]);
+    }
+    if (sortType === "sortAmountHighest") {
+        return sort_amount_highest([...transactions], walletAddress);
+    }
+    if (sortType === "sortAmountLowest") {
+        return sort_amount_lowest([...transactions], walletAddress);
+    }
+    return [];
```

Comment:
Small nitpick but I think this would be cleaner using a switch case
```suggestion
    switch (sortType) {
        case "sortDateNewest":
            return sort_date_newest([...transactions]);            
        case "sortDateOldest":
            return sort_date_oldest([...transactions]);
        case "sortAmountHighest":
            return sort_amount_highest([...transactions], walletAddress);
        case "sortAmountLowest":
            return sort_amount_lowest([...transactions], walletAddress);
        default:
            return [];            
    }
```